src/raphael_backend_flask/routes.py

```diff
@@ -171,3 +202,14 @@ def patch_user(username: str) -> ResponseReturnValue:
 def disable_user(username: str) -> ResponseReturnValue:
     disable_user_sql(username)
     return "", 200
+
+
+@routes.errorhandler(Exception)
+def handle_exception(e) -> Response:
+    if isinstance(e, FlashException):
+        flash(e.message, e.category)
+    else:
+        flash("An unexpected error occurred.", category="danger")
+        print(traceback.format_exc())
+        print(e)
+    return redirect(url_for("routes.get_home"))
```

Comment:
There are a few API routes, and these should return a 5xx status code when an exception is raised, not a 302.

I’m not sure what the preferred way to do this is – either a different exception handler, or some extra logic here that checks the request headers…

Mentioning this because the /api/register endpoint (which is usually accessed via curl) now returns a 302 when an exception is raised. That’s not right.

Reply:
[Docs suggest](https://flask.palletsprojects.com/en/2.3.x/errorhandling/#blueprint-error-handlers) this could be something like:
```python
if request.path.startswith('/api/'):
    # if it looks like an API endpoint, return JSON
    return jsonify(message=str(e)), 500
```

Alternatively, something like this maybe?

```python
if "text/html" not in request.headers["Accept"].split(","):
    # if the client didn’t request HTML, return JSON
    return jsonify(message=str(e)), 500
```

Reply:
Thanks for documenting this and for the examples solutions. I've used your first suggestion and have updated it to return JSON for requests to `/api/*`.

Reply:
Great, great – looks good 🙏🏻